projects/frontend/data-pipelines/gui/projects/data-pipelines/src/lib/components/data-job/pages/executions/execution-duration-chart/execution-duration-chart.component.ts

```diff
@@ -0,0 +1,316 @@
+            type: 'time',
+            time: {
+                unit
+            },
+            min,
+            max
+        };
+
+        this.chart.data.datasets[0].data = this.getData(min, max);
+
+        this.chart.update();
+    }
+
+    private _getMinMaxExecutionTuple(): [number, number] {
+        const executions = this.getData();
+
+        if (executions.length < 2) {
+            return [null, null];
+        }
+
+        return [executions[0].startTime, executions[executions.length - 1].startTime]
+    }
+
+    private _getTimeScaleUnit(min: number | string, max: number | string): TimeUnit {
+        if (CollectionsUtil.isNil(min) || CollectionsUtil.isNil(max)) {
+            return 'day';
+        }
+
+        const _min = CollectionsUtil.isNumber(min)
+            ? min
+            : new Date(min).getTime();
+        const _max = CollectionsUtil.isNumber(max)
+            ? max
+            : new Date(max).getTime();
+        const diff = _max - _min;
+
+        if (diff > (this._getTimeUnitMilliseconds('year') + this._getTimeUnitMilliseconds('second'))) {
+            return 'year';
+        }
+
+        if (diff > (this._getTimeUnitMilliseconds('month') + this._getTimeUnitMilliseconds('second'))) {
+            return 'month';
+        }
+
+        if (diff > (2 * this._getTimeUnitMilliseconds('week'))) {
+            return 'week';
+        }
+
+        if (diff > (this._getTimeUnitMilliseconds('day') + this._getTimeUnitMilliseconds('second'))) {
+            return 'day';
+        }
+
+        if (diff > (this._getTimeUnitMilliseconds('hour') + this._getTimeUnitMilliseconds('second'))) {
+            return 'hour';
+        }
+
+        if (diff > (this._getTimeUnitMilliseconds('minute') + this._getTimeUnitMilliseconds('millisecond'))) {
+            return 'minute';
+        }
+
+        if (diff > (this._getTimeUnitMilliseconds('second') + this._getTimeUnitMilliseconds('millisecond'))) {
+            return 'second';
+        }
+
+        return 'millisecond';
+    }
+
+    private _getTimeUnitMilliseconds(unit: 'millisecond' | 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year'): number {
+        switch (unit) {
+            case 'millisecond':
+                return 1;
+            case 'second':
+                return 1000;
+            case 'minute':
+                return 1000 * 60;
+            case 'hour':
+                return 1000 * 60 * 60;
+            case 'day':
+                return 1000 * 60 * 60 * 24;
+            case 'week':
+                return 1000 * 60 * 60 * 24 * 7;
+            case 'month':
+                return 1000 * 60 * 60 * 24 * 31;
+            case 'year':
+                return 1000 * 60 * 60 * 24 * 365;
+            default:
+                // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                console.error(`Taurus DataPipelines ExecutionDurationChartComponent unsupported time format unit ${ unit }`);
```

Comment:
I am not quite sure any automation further the line is not dependant on message formats.
No easy way to be sure of it at this point, since it's too early to modify stuff around. 

In my opinion, it's best to have there tests running, release artifacts published and consumed back.
By holding functional or I/O changes while opening, we reduce the complexity, and ease the initial adoption.

Reply:
I'm ok. 

Please, add a story/task(May be a TODO) somewhere to go over all these once we have the initial version merged in the repo.

Reply:
I'm keeping todo list to review a few occurrences of "Taurus", in the CI/CD task (added this one)